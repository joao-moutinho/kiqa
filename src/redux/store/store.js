import { combineReducers, createStore, applyMiddleware } from "redux";
import { composeWithDevTools } from "redux-devtools-extension";
import cartReducer from "../reducer";
import logReducer from "../reducer/loginred";

const rootReducers = combineReducers({ cart: cartReducer, loggedUser: logReducer });

export const store = createStore(
  rootReducers,
  composeWithDevTools(applyMiddleware())
);
