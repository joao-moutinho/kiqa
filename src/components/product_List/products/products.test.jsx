import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { Products } from ".";
import { store } from "../../../redux/store/store";


describe("Products", () => {

    const mock = [{
        id: 1,
        price : 5,
        description: "",
        brand: "",
        name: "",
        isActive: true,
        image: "  ",
        categoryName: "",
        productType: "",
        colors: [
            {
                "hexValue": "#B28378",
                "colourName": "BFF PENCIL" 
            }
        ]
    }
]
    test('render', () => {
    
        const {container} = render(<BrowserRouter> <Provider store={store}> <Products products={mock}  /> </Provider>  </BrowserRouter>);
        expect(container).toMatchSnapshot();
    })
})