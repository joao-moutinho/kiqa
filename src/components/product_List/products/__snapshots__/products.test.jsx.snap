// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products render 1`] = `
<div>
   
   
  <div
    class="products"
  >
    <div
      class="product"
    >
      <div
        class="product-image"
      >
        <img
          height="150px"
          src="  "
          width="150px"
        />
      </div>
      <div
        class="product-name"
      />
      <div
        class="product-price"
      >
        5
         â‚¬
      </div>
      <div
        class="buy-button"
      >
        <div
          class="primary-button"
        >
          <button>
            Add to Cart
          </button>
        </div>
      </div>
    </div>
  </div>
   
    
</div>
`;
